test/WebJobs.Script.Tests/UtilityTests.cs

```diff
@@ -595,5 +600,48 @@ public void TryGetFunctionName_ReturnsExpectedResult(string keyName, bool expect
                 Assert.Equal("test", functionName);
             }
         }
+
+        [Fact]
+        public void Test_LogAutorestGeneratedJson_Without_AutorestGeneratedJson()
+        {
+            string GetTemporaryDirectory()
+            {
+                string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(tempDirectory);
+                return tempDirectory;
+            }
+
+            string functionAppDirectory = GetTemporaryDirectory();
+            Utility.LogAutorestGeneratedJson(functionAppDirectory, _testLogger);
+            var allLogs = _testLogger.GetLogMessages();
+            Assert.Empty(allLogs);
+        }
+
+        [Theory]
+        [InlineData("{\"name\":\"@autorest/azure-functions-csharp\",\"version\":\"0.2.0-preview\"}", "autorest_generated.json file found generated by Autorest (https://aka.ms/stencil) | file content", LogLevel.Information)]
+        [InlineData("{\"name\":\"@autorest/azure-functions-csharp\",\"version\"\"0.2.0-preview\"}", "autorest_generated.json file found is incorrect (https://aka.ms/stencil) | exception", LogLevel.Warning)]
+        public void Test_LogAutorestGeneratedJson_With_AutorestGeneratedJson(string autorestGeneratedJsonContent, string expectedContents, LogLevel expectedLogLevel)
+        {
+            string GetTemporaryDirectory()
+            {
+                string tempDirectory = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+                Directory.CreateDirectory(tempDirectory);
+                return tempDirectory;
+            }
+
+            string functionAppDirectory = GetTemporaryDirectory();
+            File.WriteAllText(Path.Combine(functionAppDirectory, ScriptConstants.AutorestGeenratedMetadataFileName), autorestGeneratedJsonContent);
+            Utility.LogAutorestGeneratedJson(functionAppDirectory, _testLogger);
```

Comment:
Wrap the tests in try...finally - in finally block - delete the directory you created above.